packages/sdk/contractkit/src/wrappers/EpochManager.ts

```diff
@@ -0,0 +1,124 @@
+import { EpochManager } from '@celo/abis-12/web3/EpochManager'
+import BigNumber from 'bignumber.js'
+import { proxyCall } from './BaseWrapper'
+import { BaseWrapperForGoverning } from './BaseWrapperForGoverning'
+import { ValidatorGroupVote } from './Election'
+
+export enum EpochProcessStatus {
+  NotStarted,
+  Started,
+}
+
+export interface EpochProcessState {
+  status: EpochProcessStatus
+  perValidatorReward: BigNumber
+  totalRewardsVoter: BigNumber
+  totalRewardsCommunity: BigNumber
+  totalRewardsCarbonFund: BigNumber
+  toProcessGroups: number
+}
+
+/**
+ * Contract handling slashing for Validator downtime using intervals.
+ */
+export class EpochManagerWrapper extends BaseWrapperForGoverning<EpochManager> {
+  private ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+  startNextEpochProcess: () => Promise<void> = proxyCall(
+    this.contract.methods.startNextEpochProcess
+  )
+  isTimeForNextEpoch: () => Promise<boolean> = proxyCall(this.contract.methods.isTimeForNextEpoch)
+  getElected: () => Promise<string[]> = proxyCall(this.contract.methods.getElected)
+  getEpochProcessingStatus: () => Promise<EpochProcessState> = async () => {
+    const statusRaw = await this.contract.methods.epochProcessing().call()
+    return {
+      status: parseInt(statusRaw.status),
+      perValidatorReward: new BigNumber(statusRaw.perValidatorReward),
+      totalRewardsVoter: new BigNumber(statusRaw.totalRewardsVoter),
+      totalRewardsCommunity: new BigNumber(statusRaw.totalRewardsCommunity),
+      totalRewardsCarbonFund: new BigNumber(statusRaw.totalRewardsCarbonFund),
+      toProcessGroups: parseInt(statusRaw.toProcessGroups),
+    }
+  }
+  finishNextEpochProcess: (
+    groups: string[],
+    lessers: string[],
+    greaters: string[]
+  ) => Promise<void> = proxyCall(this.contract.methods.finishNextEpochProcess)
+
+  finishNextEpochProcessH: () => Promise<void> = async () => {
```

Comment:
i'm thinking that maybe the `EpochManager` has a query call that can do most of this preprocessing. So logic it's mostly on the smart contract and you then just need someone to call it.